src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirOperationDataStore.cs

```diff
@@ -5,25 +5,80 @@
+using Microsoft.Health.Fhir.SqlServer.Features.Schema.Model;
+using Microsoft.IO;
+using Newtonsoft.Json;
 
 namespace Microsoft.Health.Fhir.SqlServer.Features.Storage
 {
     public class SqlServerFhirOperationDataStore : IFhirOperationDataStore
     {
+        internal static readonly Encoding ResourceEncoding = new UnicodeEncoding(bigEndian: false, byteOrderMark: false);
+
+        private readonly SqlServerDataStoreConfiguration _configuration;
+        private readonly ILogger<SqlServerFhirOperationDataStore> _logger;
+        private readonly RecyclableMemoryStreamManager _memoryStreamManager;
+
+        public SqlServerFhirOperationDataStore(SqlServerDataStoreConfiguration configuration, ILogger<SqlServerFhirOperationDataStore> logger)
+        {
+            EnsureArg.IsNotNull(configuration, nameof(configuration));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _configuration = configuration;
+            _logger = logger;
+            _memoryStreamManager = new RecyclableMemoryStreamManager();
+        }
+
         public Task<IReadOnlyCollection<ExportJobOutcome>> AcquireExportJobsAsync(ushort maximumNumberOfConcurrentJobsAllowed, TimeSpan jobHeartbeatTimeoutThreshold, CancellationToken cancellationToken)
         {
             IReadOnlyCollection<ExportJobOutcome> returnValue = new List<ExportJobOutcome>();
             return Task.FromResult(returnValue);
         }
 
-        public Task<ExportJobOutcome> CreateExportJobAsync(ExportJobRecord jobRecord, CancellationToken cancellationToken)
+        public async Task<ExportJobOutcome> CreateExportJobAsync(ExportJobRecord jobRecord, CancellationToken cancellationToken)
         {
-            throw new NotImplementedException();
+            using (var connection = new SqlConnection(_configuration.ConnectionString))
+            {
+                await connection.OpenAsync(cancellationToken);
+
+                using (SqlCommand command = connection.CreateCommand())
+                using (var stream = new RecyclableMemoryStream(_memoryStreamManager))
+                using (var gzipStream = new GZipStream(stream, CompressionMode.Compress))
+                using (var writer = new StreamWriter(gzipStream, ResourceEncoding))
+                {
+                    var rawJobRecord = JsonConvert.SerializeObject(jobRecord);
+                    writer.Write(rawJobRecord);
+                    writer.Flush();
+
+                    stream.Seek(0, 0);
+
+                    V1.CreateExportJob.PopulateCommand(
+                        command,
+                        jobRecord.Id,
+                        jobRecord.Status.ToString(),
+                        jobRecord.QueuedTime,
+                        stream);
+
+                    int? rowVersion = (int?)await command.ExecuteScalarAsync(cancellationToken);
+
+                    // The row version should nerver be null.
+                    Ensure.That(rowVersion).IsNotNull();
```

Comment:
What would this look like when it bubbles up to the user that requested it?

Reply:
Ah, good point. I throw `OperationFailedException` in 1014e9d.